packages/cart/src/repositories/address.ts

```diff
@@ -0,0 +1,24 @@
+import { Context } from "@medusajs/types"
+import { DALUtils } from "@medusajs/utils"
+import { SqlEntityManager } from "@mikro-orm/postgresql"
+import { Address } from "@models"
+import { UpdateAddressDTO } from "../types"
+
+export class AddressRepository extends DALUtils.mikroOrmBaseRepositoryFactory<Address>(
+  Address
+) {
+  async update(
+    data: { address: Address; update: UpdateAddressDTO }[],
+    context: Context = {}
+  ): Promise<Address[]> {
+    const manager = this.getActiveManager<SqlEntityManager>(context)
+
+    const entities = data.map(({ address, update }) => {
+      return manager.assign(address, update)
+    })
+
+    manager.persist(entities)
+
+    return entities
```

Comment:
suggestion: I would prefer to call return await super.update(entities, context). The reason is that under the hood it will create manager.create to ensure that we are holding the ref of the entity. we know that here address is an instance of the entity, but as it is a different layer we should act as if we don'tknow to be future proof. wdyt?

Reply:
comment: even though the existing check will be doubled (but we are covered by the entity map pattern) it will be removed at a later point when it will be handled by the service layer. Let me know your thinking about it.